rs/nns/governance/src/lib.rs

```diff
@@ -201,6 +203,20 @@ pub fn temporarily_disable_private_neuron_enforcement() -> Temporary {
     Temporary::new(&IS_PRIVATE_NEURON_ENFORCEMENT_ENABLED, false)
 }
 
+pub fn are_set_visibility_proposals_enabled() -> bool {
+    ARE_SET_VISIBILITY_PROPOSALS_ENABLED.with(|ok| ok.get())
+}
+
+/// Only integration tests should use this.
+pub fn temporarily_enable_set_visibility_proposals() -> Temporary {
```

Comment:
How do we enforce what code can access this?  And how would integration tests access it?

Reply:
TL;DR: I don't think there is a way to do that at this time.

We could limit who can use this via bazel visibility. E.g. [this line](https://sourcegraph.com/github.com/dfinity/ic@e6d756942538836b2c83e67526efb9678a3efa19/-/blob/rs/nns/governance/BUILD.bazel?L8) could instead say

```
package(default_visibility = ["//rs/nervous_system:default_visibility"])
```

That line is used elsewhere. E.g. [here](https://sourcegraph.com/github.com/dfinity/ic@e6d756942538836b2c83e67526efb9678a3efa19/-/blob/rs/nervous_system/lock/BUILD.bazel?L3). I think once you make non-nns stuff stop depending on //rs/nns/governance, then we can make this change, but not yet.

But at a rust level, if you want to let integration tests use this, then, `pub` is unavoidable. Anyway, this code is temporary. Also, I think we can easily get used to rejecting production (i.e. non-test) code that calls functions with names of the form temporarily_(enable|disable)_*. I suppose it is possible that such code could legitimately be used in production, but it should definitely raise red flags.

Reply:
> And how would integration tests access it?

Not sure what you mean. These are used in this PR.

Reply:
One thing we can do to make it really clear that these are only for integration tests is to move these to a new file at ./only_for_integration_tests.rs.

Reply:
FWIW, having these pub fns is better than making the underlying thread_local pub. This at least limits the way that the the thread_local can be used.

Reply:
> Not sure what you mean. These are used in this PR.

Those are more like unit tests, b/c they're not working against the compiled wasm.